core/src/codegen/from_meta_impl.rs

```diff
@@ -90,16 +89,37 @@ impl<'a> ToTokens for FromMetaImpl<'a> {
                     }
                 };
 
+                let default_or_err = base.default.as_ref().map_or(
+                    quote!(::darling::export::Err(::darling::Error::too_few_items(1))),
+                    |default_expr| quote!(::darling::export::Ok(#default_expr)),
+                );
+
+                let word_or_err = variants
+                    .iter()
+                    .find_map(|variant| {
+                        if variant.word {
+                            let ty_ident = variant.ty_ident;
+                            let variant_ident = variant.variant_ident;
+                            Some(quote!(::darling::export::Ok(#ty_ident::#variant_ident)))
+                        } else {
+                            None
+                        }
+                    })
+                    .unwrap_or(quote!(::darling::export::Err(
```

Comment:
Nit: This should probably be `unwrap_or_else` to avoid an unnecessary invocation of `quote`.